README.md
# Wands
Adds magic wands to your minecraft world

*This plugin was made for Minecraft version 1.13.2, but should work on newer releases. If you encounter any problems please let me know*

## Description

Wands can be optained by killing witches and will have different functionality depending on the type of wand.
You can left click to use a wand, however make sure you left click somewhere in the air. Wands have a specific cooldown before you can use them again.

## All current Wands

### Common
#### 0 second cooldown
- **Craftsman Wand** Will open a crafting menu for you

### Uncommon
#### 5 second cooldown
- **Fireball Wand** Shoots a fireball in the direction you are looking
- **Teleport Wand** Teleports you to the block you are looking at
- **Rocket Wand** Launches you in the direction you are looking
- **Slime Wand** Does some slime stuff?

### Rare
#### 20 second cooldown
- **Trickery Wand** Gives you speed for a limited time, if you sneak in that time period you get teleported to the position you were at when using the wand
- **Ice Wand** Covers the nearby area in snow and freezes every mob inside of it

### Legendary
#### 30 second cooldown
- **Cloud Wand** Creates some temporary blocks under you so that you can walk through air
- **Earth Wand** Launches blocks and mobs nearby into the air dealing significant fall damager

### Ancient
#### 60 second cooldown
- **Summoners Wand** Summons a few angry wolfs nearby targeting random mobs or players in their area
- **Lightning Wand** Strikes lightning at the position of the block you are looking at
- **Pumpkin Wand** Spawns a bunch of flying pumpkins. These will confuse mobs and players by giving blindness and slowness effects as well as pushing the target in random directions


changelog.md
# Version 1.3.1

The next update (v1.4) will contain some new wands aswell as a bunch of code improvements as soon as I implement all of them into the new API
- Added "PluginBase" dependency. The plugin now uses that API for a bunch of stuff, it will still function the same but some parts of the plugins code have been moved to the new API and are much better written now
- Fixed bug that caused wands not to have a cooldown

# Version 1.3

- Added slime wand
- Added config file allowing the user control over some wand spawn settings
- Added permission ```wands.give```. Now only players that have this permission or are operators can recieve wands using commands
- Added ```/wands help``` command that will show a list of all commands
- Fixed coloring of pumpkin wand
- Fixed bug that caused rocket wand not to work in some situations

# Version 1.2

- Updated wand descriptions
- Wands will now spawn in different rarity types (i.e. common, uncommon, ...) These will not effect the spawn rate of the wand but will rather function as a visual indicator of its power
- Changed from a gunpowder based system to a cooldown based system. All wands will no be usable without the consumption of items. However they will need some time to recharge
- Fixed bug that caused ice wand to not work in caves
- Fixed ```/wands give <name>``` command to work with an improved way of spawning wands
- Removed ```/wands free``` command

# Version 1.1

- Added cloud, trickery and pumpkin wand
- Added particle effects to all wands
- Added ```/wands free``` command (OP only) that makes the use of wands stop taking gunpowder as fuel. This can be toggled
- Made summoners wand less likely to spawn wolfs inside of blocks
- Increased the price of fireball wand (1 -> 2)
- Fixed bug that caused wands to spawn incorrectly
- Fixed bug that caused wands to not work in caves

# Version 1.0

- Added a [Bukkit Page](https://dev.bukkit.org/projects/simple-wands)
- Official release on Bukkit Dev

# Version 0.2

- Added sound effects for all wands
- Added summoners, lightning, rocket and craftsman wands

# Version 0.1

- Made wands use different amounts of gunpowder on action
- Made wands drop after killing witches
- Added fireball, ice, earth and teleport wands
- Initial release (not publicly shared yet)


Wands/src/com/Wands/Main.java
package com.Wands;

import java.util.ArrayList;
import java.util.List;

import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.plugin.java.JavaPlugin;

import com.Wands.WandVariations.*;

public class Main extends JavaPlugin {
	
	public static double version = 1.3;
	public static boolean costEnabled = true;
	public static List<Wand> wandVariations = new ArrayList<>();
	public static FileConfiguration config;
	
	@SuppressWarnings("unused")
	@Override
	public void onEnable() {
		
		// Load the config containing plugin settings
		loadConfig();
		
		// Initiate all the wand variations
		initiateWands();
		
		// Initiate wand dropper
		new WandDropper(this);
	}
	
	void loadConfig() {
		
		// Get the config
		config = this.getConfig();
		
		// Set some default settings
		config.addDefault("Wands.Drop.Enabled", true);
		config.addDefault("Wands.Drop.Chance", 10);
		
		// Save all the settings to the config
		config.options().copyDefaults(true);
		this.saveConfig();
	}
	
	void initiateWands() {
		
		// Common
		wandVariations.add(new CraftingWand(this, 	ChatColor.RESET + "Craftsman Wand",			ChatColor.RESET + "common",				0));
		
		// Uncommon
		wandVariations.add(new FireballWand(this, 	ChatColor.AQUA + "Fireball Wand", 			ChatColor.AQUA + "uncommon",			5));
		wandVariations.add(new TeleportWand(this, 	ChatColor.AQUA + "Teleport Wand", 			ChatColor.AQUA + "uncommon",			5));
		wandVariations.add(new RocketWand(this,		ChatColor.AQUA + "Rocket Wand", 			ChatColor.AQUA + "uncommon",			5));
		wandVariations.add(new SlimeWand(this, 		ChatColor.AQUA + "Slime Wand", 				ChatColor.AQUA + "uncommon", 			5));
		
		// Rare
		wandVariations.add(new TrickeryWand(this, 	ChatColor.BLUE + "Trickery Wand", 			ChatColor.BLUE + "rare",				20));
		wandVariations.add(new IceWand(this, 		ChatColor.BLUE + "Ice Wand", 				ChatColor.BLUE + "rare",				20));
		
		// Legendary
		wandVariations.add(new WoolWand(this, 		ChatColor.GOLD + "Cloud Wand", 				ChatColor.GOLD + "legendary",			30));
		wandVariations.add(new EarthWand(this, 		ChatColor.GOLD + "Earth Wand", 				ChatColor.GOLD + "legendary",			30));
		
		// Ancient
		wandVariations.add(new SummonersWand(this, 	ChatColor.LIGHT_PURPLE + "Summoners Wand", 	ChatColor.LIGHT_PURPLE + "ancient",		60));
		wandVariations.add(new LightningWand(this, 	ChatColor.LIGHT_PURPLE + "Lightning Wand", 	ChatColor.LIGHT_PURPLE + "ancient",		60));
		wandVariations.add(new PumpkinWand(this, 	ChatColor.LIGHT_PURPLE + "Pumpkin Wand",	ChatColor.LIGHT_PURPLE + "ancient",		60));
	}
	
	@Override
	public boolean onCommand(CommandSender sender, Command cmd, String label, String[] args) {
		// Send command information to command handler
		CommandHandler.handleCommand(sender, cmd, label, args);
		return true;
	}
	
}
